analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsPattern.cs

```diff
@@ -37,18 +37,18 @@ internal sealed class IsPattern : BranchingProcessor<IIsPatternOperationWrapper>
 
     protected override ProgramState LearnBranchingConstraint(ProgramState state, IIsPatternOperationWrapper operation, bool falseBranch) =>
         state.ResolveCapture(operation.Value).TrackedSymbol() is { } testedSymbol
-        && LearnBranchingConstraint(state, operation.Pattern, falseBranch) is { } constraint
+        && LearnBranchingConstraint(state, operation.Pattern, falseBranch, state[testedSymbol]) is { } constraint
```

Comment:
The (duplicate) "hasObjectConstraint" check can be unified here:
```suggestion
        && LearnBranchingConstraint(state, operation.Pattern, falseBranch, state[testedSymbol]?.HasConstraint<ObjectConstraint>() == true) is { } constraint
```